src/Smalot/PdfParser/Page.php

```diff
@@ -216,8 +216,91 @@ public function getText(self $page = null): string
         return '';
     }
 
+    /**
+     * Return true if the current page is a (setasign\Fpdi\Fpdi) FPDI/FPDF document
+     *
+     * @return bool true is the current page is a FPDI/FPDF document
+     */
+    public function isFpdf(): bool
+    {
+        if (\array_key_exists('Producer', $this->document->getDetails()) &&
+            \is_string($this->document->getDetails()['Producer']) &&
+            str_starts_with($this->document->getDetails()['Producer'], 'FPDF')) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Return the page number of the PDF document of the page object
+     *
+     * @return int the page number
+     */
+    public function getPageNumber(): int
+    {
+        $pages = $this->document->getPages();
+        $numOfPages = \count($pages);
+        for ($pageNum = 0; $pageNum < $numOfPages; ++$pageNum) {
+            if ($pages[$pageNum] === $this) {
+                break;
+            }
+        }
+
+        return $pageNum;
+    }
+
+    /**
+     * Return the xObject if the document is from fpdf
+     *
+     * @return object The xObject for the page
+     */
+    public function getXObjectForFpdf()
```

Comment:
Can you add the return type here please? Is it `object` or does it differ from time to time?